pandas/core/frame.py

```diff
@@ -716,24 +716,24 @@ def _repr_html_(self):
     def to_string(
         self,
         buf=None,
-        columns=None,
-        col_space=None,
-        header=True,
-        index=True,
-        na_rep="NaN",
-        formatters=None,
-        float_format=None,
-        sparsify=None,
-        index_names=True,
-        justify=None,
-        max_rows=None,
-        min_rows=None,
-        max_cols=None,
-        show_dimensions=False,
-        decimal=".",
-        line_width=None,
-        max_colwidth=None,
-    ):
+        columns: Optional[List[str]] = None,
+        col_space: Optional[Union[str, int]] = None,
```

Comment:
for `to_string` only an int is permitted?

Reply:
Mostly types are copy paste from `io.formats.format`. I will have to take a closer look.

Reply:
DataFrameFormatter is shared with to_latex and to_html, so some of the types for to_string will be less permissive than DataFrameFormatter .__init__

Reply:
I will check the `to_string` method of `DataFrameFormatter` later and fix these.

Reply:
Changed `col_space` to `int`.